app/clients/sms/twilio.py

```diff
@@ -232,22 +242,64 @@ def translate_delivery_status(
         self.logger.info('Translating Twilio delivery status')
         self.logger.debug(twilio_delivery_status_message)
 
+        decoded_msg, parsed_dict = self._parse_twilio_message(twilio_delivery_status_message)
+
+        message_sid = parsed_dict['MessageSid'][0]
+        twilio_delivery_status = parsed_dict['MessageStatus'][0]
+        error_code = parsed_dict.get('ErrorCode', [])
+
+        status, status_reason = self._evaluate_status(message_sid, twilio_delivery_status, error_code)
+
+        status = SmsStatusRecord(
+            decoded_msg,
+            message_sid,
+            status,
+            status_reason,
+        )
+
+        self.logger.debug('Twilio delivery status translation: %s', status)
+
+        return status
+
+    def update_notification_status(self, message_sid: str) -> None:
```

Comment:
Should we log the notification_id here as well as the message SID?

Reply:
The whole thing is done by the message sid, the method doesn't have easy access to the notification id.

Reply:
I see, we update by referencing the `message_sid`